src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -128,13 +128,27 @@ public static void AddWebJobsScriptHost(this IServiceCollection services, IConfi
             services.AddSingleton<IFunctionMetadataManager, FunctionMetadataManager>();
             services.AddSingleton<IFunctionMetadataProvider, HostFunctionMetadataProvider>();
             services.AddSingleton<IWebFunctionsManager, WebFunctionsManager>();
-            services.AddSingleton<IInstanceManager, InstanceManager>();
             services.AddHttpClient();
             services.AddSingleton<StartupContextProvider>();
             services.AddSingleton<IFileSystem>(_ => FileUtility.Instance);
             services.AddTransient<VirtualFileSystem>();
             services.AddTransient<VirtualFileSystemMiddleware>();
 
+            if (SystemEnvironment.Instance.IsLinuxConsumptionOnAtlas())
+            {
+                System.Console.WriteLine("[TEST] Atlas Linux Consumption");
+                services.AddSingleton<LinuxInstanceManager, InstanceManager>();
+            }
+            else if (SystemEnvironment.Instance.IsLinuxConsumptionOnLegion())
+            {
+                System.Console.WriteLine("[TEST] Legion Linux Consumption");
+                services.AddSingleton<LinuxInstanceManager, LegionInstanceManager>();
+            }
+            else
+            {
+                System.Console.WriteLine("[TEST] Else");
```

Comment:
Made it default to AtlasInstanceManager since that was the previous InstanceManager injected. The IsLinuxConsumptionOnAtlas and IsLinuxConsumptionOnLegion checks should be accurate, but if they cause an issue, we should change the checks and leave the dependency injection logic